### PR TITLE
MTDSA-13849 Patched incorrect Downstream Mappings

### PR DESCRIPTION
**Pull request by sean-flynn-HMRC**

### CODE REVIEW

File: app/v1/services/CreateAmendCgtResidentialPropertyDisposalsService.scala [hunk 1]
```diff
@@ -53,9 +62,9 @@ class CreateAmendCgtResidentialPropertyDisposalsService @Inject() (connector: Cr
       "INVALID_TAX_YEAR"          -> TaxYearFormatError,
       "INVALID_CORRELATIONID"     -> StandardDownstreamError,
       "INVALID_PAYLOAD"           -> StandardDownstreamError,
-      "INVALID_DISPOSAL_DATE"     -> StandardDownstreamError,
-      "INVALID_COMPLETION_DATE"   -> StandardDownstreamError,
-      "INVALID_ACQUISITION_DATE"  -> StandardDownstreamError,
+      "INVALID_DISPOSAL_DATE"     -> RuleDisposalDateError,
+      "INVALID_COMPLETION_DATE"   -> RuleCompletionDateError,
+      "INVALID_ACQUISITION_DATE"  -> RuleAcquisitionDateAfterDisposalDateError,
```
Comment (QHMRC) [inline app/v1/services/CreateAmendCgtResidentialPropertyDisposalsService.scala:67]: hmm I added this on my one
https://github.com/hmrc/individuals-income-received-api/pull/456/files#diff-a046ce9dc91bc75d0400e8e9e9dcdae74a281be5a7d916dcbf003f89ae954498R55

Comment (sean-flynn-HMRC) [reply app/v1/services/CreateAmendCgtResidentialPropertyDisposalsService.scala:67]: I suppose it's a question for the B.As - it's essentially the same error but they've given it two slightly different names in each of our business specs. I think RULE_ACQUISITION_DATE_AFTER_DISPOSAL_DATE is more descriptive and therefore a slightly better name. If we get Fatih/Helen to approve then we can update RULE_ACQUISITION_DATE to RULE_ACQUISITION_DATE_AFTER_DISPOSAL_DATE across all specs?

Comment (QHMRC) [reply app/v1/services/CreateAmendCgtResidentialPropertyDisposalsService.scala:67]: yep that sounds like a plan

File: it/v1/endpoints/CreateAmendCgtResidentialPropertyDisposalsControllerISpec.scala [hunk 2]
```diff
@@ -635,14 +635,6 @@ class CreateAmendCgtResidentialPropertyDisposalsControllerISpec extends Integrat
             completionBeforeDisposalError,
             None,
             Some("completion date before disposal date")),
-          (
-            "AA123456A",
-            "2019-20",
-            acquisitionAfterDisposalJson,
```
Comment (QHMRC) [inline it/v1/endpoints/CreateAmendCgtResidentialPropertyDisposalsControllerISpec.scala]: Maybe you can remove the JSONs for var acquisitionAfterDisposalJson and acquisitionAfterDisposalError.
If its not being used

Comment (QHMRC) [reply it/v1/endpoints/CreateAmendCgtResidentialPropertyDisposalsControllerISpec.scala]: disposalDateJson as well

File: app/v1/requestParsers/validators/CreateAmendCgtResidentialPropertyDisposalsValidator.scala [hunk 3]
```diff
@@ -16,26 +16,17 @@
...
-  RuleGainLossError,
-  RuleIncorrectOrEmptyBodyError
-}
+import api.models.errors.{MtdError, RuleCompletionDateBeforeDisposalDateError, RuleGainLossError, RuleIncorrectOrEmptyBodyError}
 import api.requestParsers.validators.Validator
 import config.AppConfig
-import utils.CurrentDateTime
 import v1.requestParsers.validators.validations._
 import v1.models.request.createAmendCgtResidentialPropertyDisposals._
 import v1.requestParsers.validators.validations.{
```
Comment (QHMRC) [inline app/v1/requestParsers/validators/CreateAmendCgtResidentialPropertyDisposalsValidator.scala:24]: prob can get rid of these individual imports. 
Theres an "import v1.requestParsers.validators.validations._" above

**Review by QHMRC — APPROVED** [review]

**Pull request merged**